master/pkg/tasks/task.go

```diff
@@ -119,7 +120,17 @@ func (t TaskSpec) EnvVars() map[string]string {
 		"DET_CONTAINER_ID":  t.ContainerID,
 		"DET_SESSION_TOKEN": t.AllocationSessionToken,
 		"DET_USER_TOKEN":    t.UserSessionToken,
-		"DET_USER":          t.Owner.Username,
+		"DET_WORKDIR":       t.WorkDir,
+	}
+	if t.Owner != nil {
+		e["DET_USER"] = t.Owner.Username
+	}
+
+	if t.TaskContainerDefaults.NCCLPortRange != "" {
+		e["NCCL_PORT_RANGE"] = t.TaskContainerDefaults.NCCLPortRange
+	}
+	if t.TaskContainerDefaults.NCCLPortRange != "" {
+		e["GLOO_PORT_RANGE"] = t.TaskContainerDefaults.NCCLPortRange
```

Comment:
```suggestion
		e["GLOO_PORT_RANGE"] = t.TaskContainerDefaults.GlooPortRange
```

Reply:
Of course there is no such thing, which makes that a non-trivial exercise.   
I

Reply:
should be? https://github.com/determined-ai/determined/blob/7b3fee44c0846f3546f7e4bf57b231d227dcd27d/master/pkg/model/task_container_defaults.go#L23

Reply:
Ohh.  It is GLOOPortRange, not GlooPortRange -- grep didn't find what you suggested.